HACKING.md

```diff
@@ -1,5 +1,127 @@
+```
+crc start -m 12288
+```
+
+Log in as a user with the role `cluster-admin` - user `kubeadmin` is often predefined with this role.
+
+You can `export KUBECONFIG=/path/to/cluster/config` or copy/add your config to `$HOME/.kube/config`. Depending on how you set it up, you may also need to `oc login` to your cluster and add pull secrets to your credentials.
+
+## Running tests
+
+Unit tests can be run without a cluster:
+* `make`: Run unit tests, lint and format code. Do this before submitting a PR.
+* `make test-unit`: Just run unit tests.
+
+To build, deploy and test the CLO image in your own cluster:
+* `make deploy`: Build CLO image and deploy to cluster with elasticsearch.
+* `make undeploy`: Undo `make deploy`.
+
+To run CLO as a local process:
+* `make run`: Run CLO as a local process. Does not require `make deploy`.
+* `make debug`: Run CLO in the `dlv` debugger.
+
+After running `make deploy`, `make run` or `make debug`, you can run e2e tests:
+* `make test-cluster` Run all e2e tests against the running CLO.
+* `go run ./test/e2e/something`: Run tests individually, in a debugger or whatever.
+
+To do a complete deploy, test, cleanup cycle exactly as CI does (this must  start from a clean cluster e.g. after `make undeploy`):
+* `make test-olm-e2e`: Run e2e tests the same way CI does
+
+Some other useful targets (read the Makefile for more) 
+* `make image`: Build the image as $IMAGE_TAG, do not deploy it.
+
+## Explanation of `make run`
+
+You can run the CLO as a local process, outside the cluster. This is *not* the
+normal way to run an operator, and does not test all aspects of the CLO
+(e.g. problems with the CSV or OLM interactions), but it has advantages:
+
+* Fast edit/run/test cycle - runs from source code, skips some slow build/deploy steps.
+* Directly access the CLO logs on stdout/stderr
+* Control logging levels and other environment variables, e.g. `export LOG_LEVEL=debug`
+* Run CLO in a debugger, profiler or other development tools.
+
+*How it works*: An operator is actually a cluster *client*. It watches for
+changes to its own custom resources, and creates/updates other resources
+accordingly. This can all be done from outside the cluster.
+
+Examples:
+```
+make run  # Run the CLO locally
+make run-debug  # Run CLO under the dlv debugger
+LOG_LEVEL=debug make run  # Run CLO with debug logging
+RUN_CMD=foo make run # Run CLO under imaginary "foo" debugger/profiler.
+```
+
+Note `make run` will not return until you terminate the CLO.
+
+## Running a full integration test with a 4.x cluster
+
+If you have a local clone of [origin-aggregated-logging](https://github.com/openshift/origin-aggregated-logging)
+under `$GOPATH/github.com/openshift/origin-aggregated-logging` you can use its `hack/get-cluster-run-tests.sh`
+to build the logging, elasticsearch-operator, and cluster-logging-operator images; deploy a 4.x cluster;
+push the images to the cluster; deploy logging; and launch the logging CI tests.
```

Comment:
Are you using these commands to run the tests? I encounter them for the first time by reading this section

Reply:
No, that section is from before my changes. I was reluctant to remove it in case it's still useful, @vimalk78 I think you tried this path - is it worth keeping or better to drop since other methods are now documented?

Reply:
I'm going to take that section out as I suspect it's no longer needed and I"m not sure if it works.
We can put it back later if I'm wrong.